### ISSUE TITLE
P1: GPU efficiency logging — util%, temp, throttle detection, JSONL artifact

### ISSUE THREAD

**Issue by petteriTeikari** [petteriTeikari/minivess-mlops#368]
## Summary

The system monitor currently shows only VRAM allocation in the console log — a static
number that tells us nothing about whether the GPU is doing useful work. This issue
implements proper GPU efficiency monitoring with bottleneck/throttle detection.

### New fields in `ResourceSnapshot` (nvidia-smi extended query)
- `gpu_mem_bw_util_pct` — memory interface busy % (distinguishes compute vs memory bound)
- `gpu_power_w` — board power draw; well below TDP + low util = GPU starved
- `gpu_sm_clock_mhz` — SM clock; drop > 8% from baseline = thermal throttle (silent FLOPS loss)

### Two-tier console output (replacing current single 60s line)

**Every epoch — GPU efficiency line (trainer.py):**
```
f #1/3: Epoch 65/100 — train_loss=0.2061 val_loss=0.2327 lr=2.99e-05
f #1/3: [GPU]  util=87%  bw=45%  temp=71C  pwr=185W | cpu=34% | vram=5360/8192MB  clk=1980MHz  [OK]
```

**Every 10 epochs — slow resource line:**
```
f #1/3: [MEM]  ram=35.5/62.7GB(57%)  swap=6.3/16.0GB  vram=5360/8192MB  rss=8.8GB
```

**Status tokens:** `[OK]` / `[STARVE]` (gpu_util<60% AND cpu>85%) / `[THROTTLE]` (temp>87C OR sm_clock drop>8%)

### Per-epoch GPU metrics → MLflow
Log as stepped metrics alongside train/val loss:
`gpu_util_pct_mean`, `gpu_util_pct_min`, `gpu_temp_c_max`, `gpu_sm_clock_mhz_min`,
`gpu_power_w_mean`, `cpu_pct_mean`

### JSONL structured log (no regex — CLAUDE.md rule #16)
Emit `system_metrics.jsonl` alongside CSV; each line is `json.dumps(dataclasses.asdict(snapshot))`.
Upload both as MLflow artifacts at run end.

### Temp dir cleanup
`shutil.rmtree(checkpoint_dir.parent)` after each fold completes (checkpoint already in MLflow).

## Acceptance Criteria

- [ ] `ResourceSnapshot` has `gpu_mem_bw_util_pct`, `gpu_power_w`, `gpu_sm_clock_mhz`
- [ ] `nvidia-smi` query extended to collect new fields
- [ ] `SystemMonitor.epoch_summary()` drains ring buffer, returns aggregated dict
- [ ] Trainer prints `[GPU]` line every epoch using latest snapshot
- [ ] Trainer logs 6 GPU metrics to MLflow via `log_epoch_metrics`
- [ ] `[STARVE]` warning printed (once) when `gpu_util<60 AND cpu>85`
- [ ] `[THROTTLE]` warning printed (once) when `temp>87 OR sm_clock<base*0.92`
- [ ] `[MEM]` line printed every 10 epochs
- [ ] JSONL file emitted alongside CSV, both uploaded to MLflow artifacts
- [ ] `/tmp` checkpoint dirs cleaned up after fold completes
- [ ] Unit tests for all new behaviour (`test_system_monitor.py`)
- [ ] No `import re` introduced anywhere (CLAUDE.md rule #16)

## Format Design Rationale (key=value + type tags)
- `[GPU]` / `[MEM]` tags → `grep '[GPU]'` isolates GPU lines from mixed logs
- All fields `key=value` → `json.loads()` of JSONL; no regex for parsing
- ASCII only (`C` not `°C`) — no unicode rendering issues in log forwarders
- Status token at end → `grep '[THROTTLE]'` across 1000 runs finds every thermal event
